nucleus/src/main/java/za/co/absa/subatomic/domain/team/TeamMemberId.java

```diff
@@ -6,4 +6,18 @@
 public class TeamMemberId {
```

Comment:
Rather make this use Lombok. I.e. `@Value`

Then can remove the `hashCode ` and `equals` implementations.

Reply:
Unless there was a specific reason for not doing that? (Seeing that you did used Lombok elsewhere)

Reply:
This was me testing something which turned out to be a problem elsewhere then I forgot to revert it